libafl/src/observers/map.rs

```diff
@@ -444,14 +444,29 @@ where
     T: Default + Copy + 'static + Serialize + serde::de::DeserializeOwned,
 {
     /// Creates a new [`MapObserver`]
+    ///
+    /// # Safety
+    /// Will get a pointer to the map and dereference it at any point in time.
+    /// The map must not move in memory!
     #[must_use]
-    fn maybe_differential<S>(name: S, map: &'a mut [T]) -> Self
+    unsafe fn maybe_differential<S>(name: S, map: &'a mut [T]) -> Self
     where
         S: Into<String>,
     {
-        Self {
-            map: OwnedSliceMut::from(map),
+        let len = map.len();
+        let ptr = map.as_mut_ptr();
+        Self::maybe_differential_from_mut_ptr(name, ptr, len)
+    }
+
+    /// Creates a new [`MapObserver`] from an [`OwnedMutSlice`]
+    #[must_use]
+    fn maybe_differential_from_mut_slice<S>(name: S, map: OwnedMutSlice<'a, T>) -> Self
```

Comment:
These APIs are new and should be sound, as long as they are called with a `OwnedMutSlice::RefRaw`

Really, we probably want a way to `Rc<RefCell<>>` these, too(?)